ledger/cow.go

```diff
@@ -68,15 +68,8 @@ func makeRoundCowState(b roundCowParent, hdr bookkeeping.BlockHeader, prevTimest
 		lookupParent: b,
 		commitParent: nil,
 		proto:        config.Consensus[hdr.CurrentProtocol],
-		mods: ledgercore.StateDelta{
-			Accts:         make(map[basics.Address]basics.AccountData),
-			Txids:         make(map[transactions.Txid]basics.Round),
-			Txleases:      make(map[ledgercore.Txlease]basics.Round),
-			Creatables:    make(map[basics.CreatableIndex]ledgercore.ModifiedCreatable),
-			Hdr:           &hdr,
-			PrevTimestamp: prevTimestamp,
-		},
-		sdeltas: make(map[basics.Address]map[storagePtr]*storageDelta),
+		mods:         ledgercore.MakeStateDelta(&hdr, prevTimestamp),
```

Comment:
In startEvaluator, we have the paysetHint; we should use it to preinitialize the state delta allocation.
The number of transactions and the deltas are typically growing in a linear fashion, so preallocating twice as much in the array initializer would be a good idea.